### PR TITLE
Fix catalyst step6 navigation

### PR DESCRIPTION
**Pull request by stackchain**

### CODE REVIEW

**Review by wolverineks — CHANGES_REQUESTED** [review]
wrap the use effect on a custom hook and give it a good name

File: src/Catalyst/Step6.tsx [hunk 1]
```diff
@@ -151,6 +127,35 @@ const CopyButton = ({text}: {text: string}) => (
...
+        }
+      >,
+    ) => {
+      if (e.data.action.type !== 'RESET') {
+        e.preventDefault()
+      }
+    }
+    navigation.addListener('beforeRemove', handleBeforeRemove)
+    return () => {
+      navigation.removeListener('beforeRemove', handleBeforeRemove)
```
Comment (wolverineks) [inline src/Catalyst/Step6.tsx]: i think removeListener is deprecated
`const unsubscribe = navigation.addListener(...)`

Comment (wolverineks) [reply src/Catalyst/Step6.tsx]: the only thing this does is prevent going back, except for "reset"?

Comment (wolverineks) [reply src/Catalyst/Step6.tsx]: also, there is a `listeners` prop for each screen, can we use that instead?
https://reactnavigation.org/docs/navigation-events/#listeners-prop-on-screen

**Review by wolverineks — CHANGES_REQUESTED** [review]

File: src/Catalyst/Step6.tsx (hunk 1 above)
Comment (stackchain) [reply src/Catalyst/Step6.tsx:154]: Hmm weird this example was added 8 days ago https://github.com/react-navigation/react-navigation/blob/4e4935ac2584bc1a00209609cc026fa73e12c10a/example/types.check.tsx#L91
`addListener` doesn't return anything (on types) https://github.com/react-navigation/react-navigation/blob/main/packages/core/src/types.tsx#L127

Comment (stackchain) [reply src/Catalyst/Step6.tsx]: > i think removeListener is deprecated `const unsubscribe = navigation.addListener(...)`

it looks like is not marked as deprecated, anyway updating to the unsub/sub made the job easier

Comment (stackchain) [reply src/Catalyst/Step6.tsx]: > also, there is a `listeners` prop for each screen, can we use that instead? https://reactnavigation.org/docs/navigation-events/#listeners-prop-on-screen

Tried, but the event object is not coming with the same type.

**Review by wolverineks — APPROVED** [review]

**Pull request merged**